test/env/lib.rs

```diff
@@ -326,3 +473,76 @@ pub mod icrc1 {
         ledger.query("icrc1_fee", ()).await.map(|(t,)| t)
     }
 }
+
+pub fn standard_replica_burn_fn(
+    agent: Arc<Agent>,
+    canister_id: Principal,
+    amount: Nat,
+) -> BurnReturnType {
+    // This method panics if it is called but burning is not supported by the given icrc1 ledger
+    Box::pin(async move {
+        let minting_account = Decode!(
+            agent
+                .query(&canister_id, "icrc1_minting_account")
+                .with_arg(Encode!(&()).unwrap())
+                .call()
+                .await?
+                .as_slice(),
+            Option<Account>
+        )
+        .unwrap()
+        .unwrap();
+        let res =
+            Decode!(agent.update(&canister_id, "icrc1_transfer").with_arg(Encode!(&Transfer::amount_to(amount,minting_account.owner)
+).unwrap())
+.call_and_wait(waiter())
+.await.unwrap().as_slice(),Result<Nat, TransferError>)
+            .unwrap();
+        Ok(res)
+    })
+}
+
+pub fn standard_sm_burn_fn(
```

Comment:
I fell like I'm missing something. If `LedgerEnv` provides `query`and `upgrade` then why there are two different functions for burn for replica and sm? You should be able to implement the standard burn function on top of `LedgerEnv`. What am I missing?

Reply:
It depends on how the burn function is set up. For example for the cycles ledger the process of burning tokens requires more than just query and update calls. Originally we also intended to use the statemachine to be able to mint tokens/cycles too, which would definitely require more than just query and update calls.

Reply:
I see but wouldn't the `standard_*_burn_fn` be the same? Then the cycles ledger won't use the standard function.

Reply:
maybe I can make this work. I'll get back to this.

Reply:
Just using the trait for the `standard burn function` is not "object safe" when used as a field in the implementations of `LedgerEnv` so I had to find a work around. The implementations of `LedgerEnv` are now using a the same standard burn function which only relies on the trait `LedgerEnv`